src/cargo/sources/git/utils.rs

```diff
@@ -838,7 +838,7 @@ fn maybe_gc_repo(repo: &mut git2::Repository) -> CargoResult<()> {
             );
             if out.status.success() {
                 let new = git2::Repository::open(repo.path())?;
-                mem::replace(repo, new);
+                let _ = mem::replace(repo, new);
```

Comment:
Ah, why this fix instead of `*repo = new`?

Reply:
you mean remove the `mem::replace` altogether? i could do that